core/src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -988,8 +988,8 @@ private void initTriggerablesRootedBy(TreeReference rootRef,
                 if (genericRoot.isParentOf(target, false)) {
                     if (!triggeredDuringInsert.contains(triggerable)) {
                         applicable.addElement(triggerable);
+                        break;
```

Comment:
Move `break` such that it doesn't after iterating through the first target. We want to continue looping through the triggerable's targets to make sure that if one of them wasn't already processed during the repeat insert phase then we add it to process now.

Not entirely sure it is the right thing to do, but it will cast a larger net so it doesn't matter if it isn't